pkg/agent/multicast/mcast_controller_test.go

```diff
@@ -1031,6 +1032,92 @@ func TestMemberChanged(t *testing.T) {
 	}
 }
 
+func TestConcurrentEventHandlerAndWorkers(t *testing.T) {
+	c := newMockMulticastController(t, true)
+	c.ifaceStore = interfacestore.NewInterfaceStore()
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	groupIP := net.ParseIP("224.3.4.5")
+	numEvents := 250
```

Comment:
With this number, the test's logging could flood the output and even make the race stack being cut, making it hard to figure out what causes the failure. You can try removing the fix and runing the test alone. I tested a few times setting it to 10 could always trigger the race and only produced reasonable logs.

Reply:
updated